### DISCUSSION TITLE
Docs:

### DISCUSSION THREAD

**Discussion by bkdiehl** [vercel/next.js#70232] [Help]
### What is the documentation issue?

Native history api only integrates with the app router

### Is there any context that might help us understand?

It's pretty frustrating that the native history api doesn't integrate with the nextjs router unless you're using the app directory.

It is not documented that this only integrates with the app router. I just spent a lot of time upgrading my nextjs version, expecting to have this functionality in the pages router, to no avail. I certainly don't understand why the pages router can't also support this.

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#using-the-native-history-api

**Comment by samcx** [comment]
@bkdiehl When you switch the toggle to Pages Router, that section is not available.

Is there a reason why the router methods are not suitable → https://nextjs.org/docs/pages/api-reference/functions/use-router#routerreplace?

**Reply by bkdiehl** [reply]
Ah, I didn't realize there were two different sections (pages vs app router). I don't think that is obvious when coming from an internet search. 

I had assumed that, since I was upgrading to nexjts 14, that the router updates would be applied evenly to both the pages and app router. My biggest regret with using nextjs has always been the inflexibility of using the pages router. It has been both a terrible dev and user experience. I thought that finally some progress had been made and I could watch the router without causing sitewide component rerenders. As stated previously, I do not understand why the pages router can't also support this.
